engine-types/src/types.rs

```diff
@@ -1,14 +1,108 @@
-use super::{str, vec, Add, Address, String, Sub, Vec, U256};
+use crate::{str, vec, Add, Address, Display, Div, Mul, String, Sub, Vec, U256};
 use borsh::{BorshDeserialize, BorshSerialize};
 
+use crate::fmt::Formatter;
+pub use evm_core::Opcode;
+
 pub type Balance = u128;
 pub type RawAddress = [u8; 20];
 pub type RawU256 = [u8; 32]; // Big-endian large integer type.
 pub type RawH256 = [u8; 32]; // Unformatted binary data of fixed length.
 pub type EthAddress = [u8; 20];
-pub type Gas = u64;
 pub type StorageUsage = u64;
 
+#[derive(Default, Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+/// Near gas type which wraps an underlying u64.
+pub struct NearGas(u64);
+
+impl Display for NearGas {
+    fn fmt(&self, f: &mut Formatter<'_>) -> crate::fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl NearGas {
+    /// Constructs a new `NearGas` with a given u64 value.
+    pub const fn new(gas: u64) -> NearGas {
+        Self(gas)
+    }
+
+    /// Consumes `NearGas` and returns the underlying type.
+    pub fn into_u64(self) -> u64 {
```

Comment:
Explicit type notation looks good. But why not implement just `Into` trait (seems it will be a more idiomatic way)?

Reply:
You mean the `From` trait which will automatically implement `Into` as implementing `Into` directly isn't recommended. Originally was say, as per discussions above but Michael believes and I do agree with our application, we should be explicit and use explicit conversions where possible. That way you can't just do `.into()` lazily.